dll/win32/shell32/CNewMenu.cpp

```diff
@@ -463,121 +463,145 @@ HRESULT CNewMenu::CreateNewFolder(LPCMINVOKECOMMANDINFO lpici)
         return E_FAIL;
 
     /* Show and select the new item in the def view */
-    SelectNewItem(SHCNE_MKDIR, SHCNF_PATHW, wszName);
+    SelectNewItem(SHCNE_MKDIR, SHCNF_PATHW, wszName, TRUE);
 
     return S_OK;
 }
 
-HRESULT CNewMenu::CreateNewItem(SHELLNEW_ITEM *pItem, LPCMINVOKECOMMANDINFO lpcmi)
+HRESULT CNewMenu::NewItemByCommand(SHELLNEW_ITEM *pItem, LPCWSTR wszPath)
+{
+    WCHAR wszBuf[MAX_PATH], wszShortPath[MAX_PATH];
+    LPWSTR Ptr, pwszCmd;
+    WCHAR wszTemp[MAX_PATH];
+    STARTUPINFOW si;
+    PROCESS_INFORMATION pi;
+
+    if (!GetShortPathNameW(wszPath, wszShortPath, _countof(wszShortPath)))
```

Comment:
This is going to be buggy in general.

Reply:
`rundll32` requires short path name.

Reply:
I'm on it.

Reply:
I've tested, and it works fine with long path names with or without spaces.